src/controllers/auth.controller.js

```diff
@@ -47,13 +67,71 @@ const verifyEmail = catchAsync(async (req, res) => {
   res.status(httpStatus.NO_CONTENT).send();
 });
 
+// PUT: Update user details by ObjectID
+const updateuser = catchAsync(async (req, res) => {
+  try {
+    // Find user by ObjectID
+    const userExist = await userService.ExistUser(req.params.userId);
+    (!userExist) ? res.status(httpStatus.NOT_FOUND).send({ msg: "User are not found" }) : 0;
+
+    let payload = req.body
+    if (payload.mobile_no) {
+      const MobileNoExist = await userService.getUserByMobileNo(payload.mobile_no);
+      (MobileNoExist) ? res.status(httpStatus.UNPROCESSABLE_ENTITY).send({ msg: "MobileNo already exist" }) : 0
+      payload.isMobileNoVerified = false
+      // generate OTP
+      var otp = await tokenService.generateVerifyOtp(userExist);
+    }
+
+    (req.file) ? payload.userImage = req.file.filename : 0
+
+    const update = await userService.updateuser(req.params.userId, payload);
+    res.status([200]).send({ msg: 'user update successfully', data: update, otp: otp });
+
+  }
+  catch (error) {
+    res.status(httpStatus.BAD_REQUEST).send({ success: false, msg: "something went's wrong", error: error });
+  }
+});
+
+// DELETE: show all data from database(Soft delete)
+const deleteUser = catchAsync(async (req, res) => {
+  const ExistUser = await userService.getUserById(req.params.userId);
+  if (!ExistUser) {
+    throw new ApiError(httpStatus.NOT_FOUND, 'This User are not found');
+  } else {
+    const result = await userService.deleteUserById(req.params.userId);
+    res.status(200).json({ success: true, mdg: 'Deleted Successfully', data: result });
+  }
+});
+
+/**
+ * List of Patient
+ */
+const userList = catchAsync(async (req, res) => {
```

Comment:
Is this really needed? Thanks

Reply:
One time passwords? It doesn’t really seem that necessary